### PR TITLE
Replace controller test with request test

### PR DESCRIPTION
**Pull request by padv2010**
## Description

Follows work on #51 and https://github.com/DFE-Digital/international-teacher-relocation-payment/pull/51#discussion_r1228180249. 

Removes controller test and replaces it with a request test.

### TODO Next

1. Remove dependency from the session to persist attributes
2. Update integration test [to remove session mock](https://github.com/DFE-Digital/international-teacher-relocation-payment/pull/59/files#diff-d48e4e211b19e3030ddbf83b9d225932a211356cf1c692866a0d772b08650edeR68-R74).

### CODE REVIEW

**Comment by padv2010** [comment]
[Trainee route: start of contract with the school](https://trello.com/c/Fl92ae9o/75-trainee-route-start-of-contract-with-the-school)

File: spec/requests/contract_start_date_question_spec.rb [hunk 1]
```diff
@@ -2,45 +2,45 @@
 
 module Applicants
   RSpec.describe ContractStartDatesController do
```
Comment (duncanjbrown) [inline spec/requests/contract_start_date_question_spec.rb:4]: nit: maybe rename this spec

File: spec/requests/contract_start_date_question_spec.rb [hunk 2]
```diff
@@ -58,12 +58,21 @@ module Applicants
...
           end
         end
       end
+
+      def set_user_type(route_type)
+        # TODO: Remove this stub when we have a better way of storing the route type on the DB
+        # The current implementation of the application route is to store it in the session,
+        # this is not ideal and should be changed to be stored in the database. For now we are
+        # stubbing the session to return the route type, but this is temporary.
+        allow_any_instance_of(described_class).to receive(:session).and_return({ "application_route" => route_type })
```
Comment (duncanjbrown) [inline spec/requests/contract_start_date_question_spec.rb:73]: 👍

**Review by duncanjbrown — APPROVED** [review]
Brilliant, thank you!

File: spec/requests/contract_start_date_question_spec.rb (hunk 1 above)
Comment (padv2010) [reply spec/requests/contract_start_date_question_spec.rb]: Addressed in [96dc741](https://github.com/DFE-Digital/international-teacher-relocation-payment/pull/59/commits/96dc7415240767628eb66a25f0e7620c55c7210e)

**Pull request merged**